### PR TITLE
Documentation updates for NServiceBus.Persistence.AzureTable v5

### PR DESCRIPTION
**Pull request by lailabougria**

### CODE REVIEW

File: persistence/azure-table/transactions_sharing_astp_[5,).partial.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+The `TestableAzureTableStorageSession` class in the `NServiceBus.Testing` namespace has been provided to facilitate [testing a handler](/nservicebus/testing/) that utilizes the shared transaction feature.
+
+### With dependency injection
+
+For custom types that require access to the shared `List<TableTransactionAction>`:
+
+snippet: TransactionalBatchRegisteredWithDependencyInjectionResolvedInCustomType
+
+And alternatively to using the the extension method `IMessageHandlerContext.SynchronizedStorageSession.GetSharedTransactionalBatch()`:
```
Comment (danielmarbach) [inline persistence/azure-table/transactions_sharing_astp_[5,).partial.md:21]: I think something is off here and in the previous documentation. This method does not exist and should probably be https://github.com/Particular/NServiceBus.Persistence.AzureTable/blob/master/src/NServiceBus.Persistence.AzureTable/SynchronizedStorage/SynchronizedStorageSessionExtensions.cs#L13

Comment (danielmarbach) [reply persistence/azure-table/transactions_sharing_astp_[5,).partial.md]: Pushed

File: Snippets/ASP/ASTP_5/Usage.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        persistence.DefaultTable("TableName");
+
+        // make sure the table name specified in the DefaultTable exists when calling DisableTableCreation
+        endpointConfiguration.EnableInstallers();
+        persistence.DisableTableCreation();
+
+        #endregion
+    }
+
+    void ConfigureThrottlingSettingsOnStorageTableClient(EndpointConfiguration endpointConfiguration, TableServiceClient tableServiceClient)
```
Comment (abparticular) [inline Snippets/ASP/ASTP_5/Usage.cs:168]: Is this snippet a wip? The text in `index_ratelimiting_astp_[5,).partial.md` suggests the retry policy can be adjusted by using `TableClientOptions`, and it looks like that was started but not finished?

File: persistence/azure-table/configuration_code_astp_[5,).partial.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+ * `ConnectionString`: Sets the connection string for the storage account to be used for storing saga information.
+ * `UseTableServiceClient`: Allows to set a fully pre-configured Table Service client instead of using a connection string.
+
+#### Saga compatibility configuration
+
+snippet: AzurePersistenceSagasCompatibility
+
+The following settings are available for changing the behavior of saga persistence compatibility section:
+
+ * `EnableSecondaryKeyLookupForSagasCorrelatedByProperties`: To enable the compatibility mode and force the persister to find sagas by the secondary index property. This is only useful while there continue to exist sagas that are using the secondary index property `NServiceBus_2ndIndexKey`.
```
Comment (abparticular) [inline persistence/azure-table/configuration_code_astp_[5,).partial.md:20]:
```suggestion
 * `EnableSecondaryKeyLookupForSagasCorrelatedByProperties`: To enable the compatibility mode and force the persister to find sagas by the secondary index property. This is only useful while there are still sagas that use the secondary index property `NServiceBus_2ndIndexKey`.
```

File: persistence/azure-table/performance-tuning_table-scanning_astp_[5,).partial.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+## Compatibility mode
+
+By default the persister runs with the [compatibility mode](/persistence/azure-table/configuration.md#saga-configuration) disabled. If the compatibility mode was manually enabled, it's important to validate whether there are sagas in the system that require the compatibility mode to be enabled.
```
Comment (abparticular) [inline persistence/azure-table/performance-tuning_table-scanning_astp_[5,).partial.md:3]:
```suggestion
By default the persister runs with the [compatibility mode](/persistence/azure-table/configuration.md#saga-configuration-saga-compatibility-configuration) disabled. If the compatibility mode was manually enabled, it's important to validate whether there are sagas in the system that require the compatibility mode to be enabled.
```

File: persistence/azure-table/transactions_intro_astp_[5,).partial.md [hunk 5]
```diff
@@ -0,0 +1 @@
+By default, the persistence does not attempt to atomically commit saga data and/or business data and uses the saga id as partition key to store sagas. Through the use of the [TableBatchOperation API](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.table.tablebatchoperation?view=azure-dotnet), saga data and/or business data can be atomically committed if everything is stored in the same partition within the same table.
```
Comment (abparticular) [inline persistence/azure-table/transactions_intro_astp_[5,).partial.md:1]: This partial is still referring to `Microsoft.Azure.Cosmos.Table.TableBatchOperation`

File: persistence/upgrades/asp-4to5.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+---
+
+
+## Switching to `Azure.Data.Tables`
+
+This version introduces support for [`Azure.Data.Tables`](https://www.nuget.org/packages/Azure.Data.Tables/) and moves away from the deprecated [`Microsoft.Azure.Cosmos.Table`](https://www.nuget.org/packages/Microsoft.Azure.Cosmos.Table/1.0.8) package.
+
+For an overview of the SDK changes, refer to the official Azure [migration guide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/tables/Azure.Data.Tables/MigrationGuide.md).
+
+In brief though, instances of `CloudTableClient` need to be replaced with `TableServiceClient` and `TableEntity` needs to be replaced with `ITableEntity` when declaring saga entities.
```
Comment (abparticular) [inline persistence/upgrades/asp-4to5.md:21]: Should we mention IgnoreDataMember/IgnoreProperty change?

**Review by abparticular — COMMENTED** [review]
For ASQ we also updated the relevant text file in components/core-dependencies but I'm not sure if that's still necessary or not?

File: Snippets/ASP/ASTP_5/Usage.cs (hunk 2 above)
Comment (lailabougria) [reply Snippets/ASP/ASTP_5/Usage.cs:168]: Good catch Andreas, this just needs to go, as it's part of the SDK api.

**Review by abparticular — APPROVED** [review]

File: Snippets/ASP/ASTP_5/ASTP_5.csproj [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>net48</TargetFramework>
+    <LangVersion>7.3</LangVersion>
+    <RootNamespace>ASTP_5</RootNamespace>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus.Persistence.AzureTable" Version="5.0.0-rc.1" />
+    <PackageReference Include="NServiceBus" Version="8.0.0" />
```
Comment (danielmarbach) [inline Snippets/ASP/ASTP_5/ASTP_5.csproj:9]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.*" />
```

File: persistence/azure-table/configuration_code_astp_[5,).partial.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+snippet: AzurePersistenceSubscriptionsCustomization
+
+The following settings are available for changing the behavior of subscription persistence:
+
+ * `ConnectionString`: Sets the connection string for the storage account to be used for storing subscription information.
+ * `UseTableServiceClient`: Allows to set a fully pre-configured Table Service client instead of using a connection string.
+
+### Configuring a Table Service Client Provider
+
+A fully preconfigured TableServiceClient can be registered in the container through a custom provider.
```
Comment (danielmarbach) [inline persistence/azure-table/configuration_code_astp_[5,).partial.md:35]: I see a mix of Table Service Client versus TableServiceClient. Should we align or does the mix follow a pattern? Given this is a specific type should we use the type indicator TableServiceClient?

File: persistence/upgrades/asp-4to5.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
+---
+title: Azure Storage Persistence Upgrade Version  to 5
```
Comment (danielmarbach) [inline persistence/upgrades/asp-4to5.md:2]:
```suggestion
title: Azure Storage Persistence Upgrade Version 4 to 5
```

File: persistence/upgrades/asp-4to5.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+summary: Instructions on how to migrate from Azure Table Persistence v4 to Azure Table Persistence version 5
```
Comment (danielmarbach) [inline persistence/upgrades/asp-4to5.md:3]:
```suggestion
summary: Instructions on how to migrate from Azure Table Persistence version 4 to 5
```

File: samples/azure/azure-table/saga-transactions/ASTP_5/Server/Server.csproj [hunk 11]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net6.0;net48</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="8.0.0" />
```
Comment (danielmarbach) [inline samples/azure/azure-table/saga-transactions/ASTP_5/Server/Server.csproj:8]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.*" />
```

File: samples/azure/azure-table/saga-transactions/ASTP_5/SharedMessages/SharedMessages.csproj [hunk 12]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net6.0;net48</TargetFrameworks>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="8.0.0" />
```
Comment (danielmarbach) [inline samples/azure/azure-table/saga-transactions/ASTP_5/SharedMessages/SharedMessages.csproj:7]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.*" />
```

File: samples/azure/azure-table/simple/ASTP_5/Server/Server.csproj [hunk 13]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net6.0;net48</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="8.0.0" />
```
Comment (danielmarbach) [inline samples/azure/azure-table/simple/ASTP_5/Server/Server.csproj:8]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.*" />
```

File: samples/azure/azure-table/simple/ASTP_5/SharedMessages/SharedMessages.csproj [hunk 14]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net6.0;net48</TargetFrameworks>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="8.0.0" />
```
Comment (danielmarbach) [inline samples/azure/azure-table/simple/ASTP_5/SharedMessages/SharedMessages.csproj:7]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.*" />
```

File: samples/azure/azure-table/table/ASTP_5/Server/Server.csproj [hunk 15]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net6.0;net48</TargetFrameworks>
+    <OutputType>Exe</OutputType>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="8.0.0" />
```
Comment (danielmarbach) [inline samples/azure/azure-table/table/ASTP_5/Server/Server.csproj:8]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.*" />
```

File: samples/azure/azure-table/table/ASTP_5/SharedMessages/SharedMessages.csproj [hunk 16]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net6.0;net48</TargetFrameworks>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="8.0.0" />
```
Comment (danielmarbach) [inline samples/azure/azure-table/table/ASTP_5/SharedMessages/SharedMessages.csproj:7]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.*" />
```

File: samples/azure/azure-table/transactions/ASTP_5/SharedMessages/SharedMessages.csproj [hunk 17]
```diff
@@ -0,0 +1,9 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>net6.0;net48</TargetFrameworks>
+    <LangVersion>7.3</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="NServiceBus" Version="8.0.0-rc.3" />
```
Comment (danielmarbach) [inline samples/azure/azure-table/transactions/ASTP_5/SharedMessages/SharedMessages.csproj:7]:
```suggestion
    <PackageReference Include="NServiceBus" Version="8.*" />
```

**Review by danielmarbach — APPROVED** [review]
One last nit, but that can also be sorted as another PR if we want

**Review by abparticular — APPROVED** [review]

**Pull request merged**